### PR TITLE
Added information on Variational Auto Encoders.

### PR DESCRIPTION
**Pull request by codingblazes**
Variational Autoencoders (VAEs) have emerged as one of the most popular approaches to unsupervised learning of complicated distributions. VAE's are appealing because they are built on top of standard function approximators (neural networks), and can be trained with stochastic gradient descent.

### CODE REVIEW

**Review by ashwinpn — COMMENTED** [review]
Add information on convex optimization and applied mathematics structures used in Deep Learning.
